pandas/io/pytables.py

```diff
@@ -1092,6 +1092,8 @@ def remove(self, key: str, where=None, start=None, stop=None):
         except KeyError:
             # the key is not a valid store, re-raising KeyError
             raise
+        except AssertionError:
```

Comment:
what are these from?

Reply:
these are unrelated to most of the PR, but related to the couple-weeks-ago push to get rid of `except Exception`.  In cases where we can't (yet) get rid of `except Exception`, at least re-raising assertion errors is useful to debugginig

Reply:
if this is a sticking point im happy to revert and do this elsewhere

Reply:
no its fine, if possible can you add a comment indicating where its raised from